treewide: fix double quoted strings in meta.description

siraben opened the pull request:
###### Motivation for this change
Replace double single quotes `''` with single double quotes `"` in various fields:
- description

###### Things done
```ShellSession
$  git ls-files | grep pkgs | xargs sed -i -e -E "s#description.*''(.*)''#description = \"\1\"#g"
```

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

lourkeur commented on nixos/modules/services/scheduling/fcron.nix:
```diff
@@ -73,7 +73,7 @@ in
         type = types.lines;
         default = "";
-        description = ''The "system" crontab contents.'';
+        description = "The "system" crontab contents.";
```
```suggestion
        description = "The \"system\" crontab contents.";
```
That should fix the evaluation error.

SuperSandro2000 commented:
> ###### Things done
> ```
> $  git ls-files | grep pkgs | xargs sed -i -e -E "s#description.*''(.*)''#description = \"\1\"#g"
> ```

I am going to include this in nüchtos.

kvtb commented:
https://github.com/siraben/nixpkgs/pull/4

siraben commented:
Rebased on `staging` again because of another treewide PR (#108022).

Ericson2314 commented:
@siraben for this I would try to get a rebuild-0 if possible? Or rather if there are intentional hash-changes, do those as separate commits.

SuperSandro2000 commented:
@siraben please fix the merge conflict and try to find the packages that cause a rebuild.

siraben commented:
@Ericson2314 Looking at some of the hash changes, I'm not sure why they are changing, e.g.

```diff
modified   pkgs/development/compilers/stalin/default.nix
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ ncompress libX11 ];
 
-  buildPhase = '' ./build '';
+  buildPhase = " ./build ";
 
   installPhase = ''
     mkdir -p "$out/bin"
modified   pkgs/development/compilers/swift/default.nix
@@ -321,7 +321,7 @@ stdenv.mkDerivation {
   '';
 
   # Hack to avoid build and install directories in RPATHs.
-  preFixup = ''rm -rf $SWIFT_BUILD_ROOT $SWIFT_INSTALL_DIR'';
+  preFixup = "rm -rf $SWIFT_BUILD_ROOT $SWIFT_INSTALL_DIR";
 
   meta = with stdenv.lib; {
     description = "The Swift Programming Language";
modified   pkgs/development/haskell-modules/hackage-packages.nix
@@ -99501,7 +99501,7 @@ self: {
          sed -i "s|\"-s\"|\"\"|" ./Setup.hs
          sed -i "s|numJobs (bf bi)++||" ./Setup.hs
        '';
-       preBuild = ''export LD_LIBRARY_PATH=`pwd`/dist/build''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH'';
+       preBuild = "export LD_LIBRARY_PATH=`pwd`/dist/build\${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH";
        description = "Grammatical Framework";
        license = "unknown";
        hydraPlatforms = stdenv.lib.platforms.none;
```

siraben commented:
@kvtb Thank you for the changes, but I will revert them here to focus on changing one-line descriptions which will ensure no rebuilds. I moved the other changes into a different PR.

siraben committed:
treewide: fix double quoted strings in meta.description

siraben commented:
@Ericson2314 rebuild 0.

SuperSandro2000 commented:
```diff
-       preBuild = ''export LD_LIBRARY_PATH=`pwd`/dist/build''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH'';
+       preBuild = "export LD_LIBRARY_PATH=`pwd`/dist/build\${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH";
```

I think the '' in the middle causes the hash change.

siraben commented:
> I think the '' in the middle causes the hash change.

But these two lines evaluate to the same string, hm.

```
nix-repl> ''export LD_LIBRARY_PATH=`pwd`/dist/build''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH''
"export LD_LIBRARY_PATH=`pwd`/dist/build${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"

nix-repl> "export LD_LIBRARY_PATH=`pwd`/dist/build\${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
"export LD_LIBRARY_PATH=`pwd`/dist/build${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
```

SuperSandro2000 commented:
> But these two lines evaluate to the same string, hm.

Then I have no idea. It would be great if you could do a separate PR with that change so it can be easily reviewed.

kvtb commented:
This must be a Nix bug to be reported and investigated
 
> > I think the '' in the middle causes the hash change.
> 
> But these two lines evaluate to the same string, hm.
> 
> ```
> nix-repl> ''export LD_LIBRARY_PATH=`pwd`/dist/build''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH''
> "export LD_LIBRARY_PATH=`pwd`/dist/build${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
> 
> nix-repl> "export LD_LIBRARY_PATH=`pwd`/dist/build\${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
> "export LD_LIBRARY_PATH=`pwd`/dist/build${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
> ```

siraben commented:
@kvtb due to many treewide PRs, could you re-run your scripts to find more instances of double quoting and open a new PR?

kvtb commented:
> @kvtb due to many treewide PRs, could you re-run your scripts to find more instances of double quoting and open a new PR?

yes. it is interesting to find the root cause.
there is either Nix parser bug or invisible change (like `\r\n` -> `\n`)

kvtb commented:
> @Ericson2314 Looking at some of the hash changes, I'm not sure why they are changing, e.g.
> 
> ```diff
> modified   pkgs/development/compilers/stalin/default.nix
> @@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
>  
> -  buildPhase = '' ./build '';
> +  buildPhase = " ./build ";
> ```

These are not equal

kvtb commented:
@siraben so you can cherry-pick https://github.com/nixos/nixpkgs/commit/891e54dc1143a545836eff27ebe091488dd2523d, it is against the current master

siraben commented:
@kvtb I can't seem to checkout that commit, is it on a branch somewhere?

siraben commented:
@kvtb nevermind, I was able to get it by adding .patch to the URL and refreshing a few times.
